### ISSUE TITLE
false positive: logger.info flagged as expensive

### ISSUE THREAD

**Issue by BertKoor** [jborgers/PMD-jPinpoint-rules#158]
Source code:

````
    private static void logApplicationStartup(Environment env) {
        LOGGER.info("Application '{}' has started!", env.getProperty("spring.application.name"));
    }
````
raises condition `pmd:UnconditionalOperationOnLogArgument`

> **Operation is executed regardless of log level and can be expensive.**
> Problem: An operation is executed regardless of log level. This could be much processing while the result is typically not used. Detected are obj.toString() and operations with one or more arguments except usually cheap obj.get(arg). Solution: Execute the operation only conditionally and utilize SLF4J formatting with {}-placeholders. (jpinpoint-rules)

I understand this rule for logging at levels DEBUG & TRACE, but logging on INFO level is usually enabled as well. Therefore imho this is a false positive.

**Comment by BertKoor** [comment]
Found a similar case with slightly different message.
Source code:
````
myCollection.stream().filter(Objects::nonNull).distinct().forEach(value ->
         LOGGER.info("value: [" + value.getAttrib1() + ", " + value.getAttrib2()));
````
raises condition `pmd:UnconditionalConcatInLogArgument`

> **String concatenation (+) is executed regardless of log level and can be expensive**
> Problem: String concatenation (+) is executed regardless of log level and can be expensive. Solution: Use SLF4J formatting with {}-placeholders or log and format conditionally. (jpinpoint-rules)
